asllib/doc/ASLTypingReference.tex

```diff
@@ -10754,456 +13968,944 @@ \section{TypingRule.BitFieldsEqual \label{sec:TypingRule.BitFieldsEqual}}
+\inferrule[different\_lengths]{
+  \equallength(\bfone, \bftwo) \typearrow \False
+}{
+  \bitfieldsequal(\tenv, \bfone, \bftwo) \typearrow \False
+}
+\and
+\inferrule[same\_lengths]{
+  \equallength(\bfone, \bftwo) \typearrow \True\\
+  i\in\listrange(\bfone): \bitfieldequal(\tenv, \bfone[i], \bftwo[i]) \typearrow \vb_i\\\\
+  \vb \eqdef \bigwedge_{i\in\listrange(\bfone)} \vb_i
+}{
+  \bitfieldsequal(\tenv, \bfone, \bftwo) \typearrow \vb
+}
+\end{mathpar}
+\end{emptyformal}
+
+\section{TypingRule.BitFieldEqual \label{sec:TypingRule.BitFieldEqual}}
+The function
+\[
+  \bitfieldequal(\overname{\staticenvs}{\tenv} \aslsep \overname{\bitfield}{\bfone} \aslsep \overname{\bitfield}{\bftwo})
+  \aslto \overname{\{\True, \False\}}{\vb} \cup \overname{\TTypeError}{\TypeErrorConfig}
+\]
+conservatively tests whether the bitfield $\bfone$ is equivalent to the bitfield $\bftwo$ in environment $\tenv$
+and yields the result in $\vb$.  The result is a type error, if one is detected.
+
+One of the following applies:
+\begin{itemize}
+  \item All of the following apply (\textsc{different\_labels}):
+  \begin{itemize}
+    \item the AST labels of $\bfone$ and $\bftwo$ are different;
+    \item $\vb$ is $\False$.
+  \end{itemize}
+
+  \item All of the following apply (\textsc{bitfield\_simple}):
+  \begin{itemize}
+    \item $\bfone$ is a simple bitfield with name $\nameone$ and slices $\slicesone$, that is, \\ $\BitFieldSimple(\nameone, \slicesone)$;
+    \item $\bftwo$ is a simple bitfield with name $\nametwo$ and slices $\slicestwo$, that is, \\ $\BitFieldSimple(\nametwo, \slicestwo)$;
+    \item checking whether $\nameone$ is equal to $\nametwo$ yields $\vbone$;
+    \item testing whether $\slicesone$ and $\slicestwo$ are equivalent in $\tenv$ yields $\vbtwo$ \ProseOrTypeError;
+    \item $\vb$ is $\True$ if and only if both $\vbone$ and $\vbtwo$ are $\True$.
+  \end{itemize}
+
+  \item All of the following apply (\textsc{bitfield\_nested}):
+  \begin{itemize}
+    \item $\bfone$ is a nested bitfield with name $\nameone$, slices $\slicesone$, and nested bitfields $\bfoneone$, that is,
+          $\BitFieldNested(\nameone, \slicesone, \bfoneone)$;
+    \item $\bftwo$ is a nested bitfield with name $\nametwo$, slices $\slicestwo$, and nested bitfields $\bftwoone$, that is,
+          $\BitFieldNested(\nametwo, \slicestwo, \bftwoone)$;
+    \item checking whether $\nameone$ is equal to $\nametwo$ yields $\vbone$;
+    \item testing whether $\slicesone$ and $\slicestwo$ are equivalent in $\tenv$ yields $\vbtwo$ \ProseOrTypeError;
+    \item testing whether the bitfields $\bfoneone$ and $\bftwoone$ are equivalent in $\tenv$ yields $\vbtwo$ \ProseOrTypeError;
+    \item $\vb$ is $\True$ if and only if both $\vbone$ and $\vbtwo$ are $\True$.
+  \end{itemize}
+
+  \item All of the following apply (\textsc{bitfield\_typed}):
+  \begin{itemize}
+    \item $\bfone$ is a typed bitfield with name $\nameone$, slices $\slicesone$, and type $\vtone$, that is,
```

Comment:
slice (no s)?

Reply:
it's actually a list of slices. I added "list of ".